src/schemathesis/cli/junitxml.py

```diff
@@ -0,0 +1,42 @@
+import platform
+from typing import List, Optional
+
+import attr
+from click.utils import LazyFile
+from junit_xml import TestCase, TestSuite, to_xml_report_file
+
+from ..models import Status
+from ..runner import events
+from .handlers import EventHandler, ExecutionContext
+from .output import default
+
+
+@attr.s(slots=True)
+class JunitXMLHandler(EventHandler):
+    file_handle: LazyFile = attr.ib()
+    test_cases: List = attr.ib(factory=list)
+    start_time: Optional[float] = attr.ib(default=None)
+
+    def handle_event(self, context: ExecutionContext, event: events.ExecutionEvent) -> None:
+        if isinstance(event, events.Initialized):
+            self.start_time = event.start_time
+        if isinstance(event, events.BeforeExecution):
+            self.test_cases.append(TestCase(f"{event.method} {event.path}"))
+        if isinstance(event, events.AfterExecution):
+            self.test_cases[-1].elapsed_sec = event.elapsed_time
+            if event.status == Status.failure:
+                checks = default._get_unique_failures(event.result.checks)
```

Comment:
It would be better to have `_get_unique_failures` outside of both handlers, so `junitxml` will not be dependent on `default`